src/app.rs

```diff
@@ -54,47 +70,68 @@ impl<'a> App<'a> {
     pub fn new(title: &'a str, x_bounds: [f64; 2], y_bounds: [f64; 2]) -> App<'a> {
         App {
             title,
-            tabs: TabsState::new(vec!["Graham Scan", "Another algorithm"]),
+            tabs: TabsState::new(vec![
+                Tab::new(AlgorithmWrapper::GrahamScan(GrahamScan::new())),
+                Tab::new(AlgorithmWrapper::GrahamScan(GrahamScan::new())),
+            ]),
             input_mode: InputMode::Normal,
             input: String::new(),
-            algorithm: GrahamScan::new(),
-            step: 0,
-            max_steps: None,
-            point_amount: None,
             x_bounds,
             y_bounds,
             should_quit: false,
         }
     }
 
+    pub fn get_current_tab(&self) -> &Tab {
```

Comment:
Do not use indexing with `[]` as this leads to panics if the index is out of bounds. Handle a possible error here and use a method like `array.get(index)`.